hot.h

```diff
@@ -0,0 +1,297 @@
+    {
+        return 1;
+    }
+};
+
+template <typename T1, typename T2>
+class HotOptions
+    : public GenericSpanOptions<HotOptions<T1, T2>, HotContext<T1, T2>, HotAdapter<T1, T2> > {
+  public:
+    int setOperationImp(const StringRef&)
+    {
+        return 0;
+    }
+    int setStartTimeImp(const uint64_t)
+    {
+        return 0;
+    }
+    int setReferenceImp(const SpanReferenceType::Value, const HotContext<T1, T2>&)
+    {
+        return 0;
+    }
+
+    template <typename T>
+    int setTagImp(const StringRef&, const T&)
+    {
+        return 0;
+    }
+};
+
+template <typename T1, typename T2>
+class HotSpan : public GenericSpan<HotSpan<T1,T2>, HotContext<T1,T2>, HotAdapter<T1,T2> > {
+    typedef typename T1::Span SpanT1;
+    typedef typename T2::Span SpanT2;
+
+    // Note: Might be NULL
+    SpanT1 * span1;
+    // Note: Might be NULL
+    SpanT2 * span2;
+
+    friend class HotTracer<T1, T2>;
+
+    HotSpan(SpanT1 * span1_, SpanT2 * span2_)
+        : span1(span1_)
+        , span2(span2_)
+    {
+    }
+
+  public:
+
+    const HotContext<T1,T2>* contextImp() const
+    {
+        return &m_context;
+    }
+
+    int setOperationImp(const StringRef&)
+    {
+        return 0;
+    }
+    int setBaggageImp(const StringRef&, const StringRef&)
+    {
+        return 0;
+    }
+
+    int getBaggageImp(const StringRef&, std::string*) const
+    {
+        return 1;
+    }
+    int getBaggageImp(const StringRef&, std::vector<std::string>*) const
+    {
+        return 1;
+    }
+
+    int finishImp()
+    {
+        return 0;
+    }
+    int finishImp(const uint64_t)
+    {
+        return 0;
+    }
+
+    template <typename T>
+    int tagImp(const StringRef&, const T&)
+    {
+        return 0;
+    }
+
+    template <typename T>
+    int logImp(const StringRef&, const T&)
+    {
+        return 0;
+    }
+
+    template <typename T>
+    int logImp(const StringRef&, const T&, const uint64_t)
+    {
+        return 0;
+    }
+
+    HotContext<T1,T2> m_context;
+};
+
+template <typename T1, typename T2>
+class HotTracer : public GenericTracer<HotTracer<T1, T2>,
+                                       HotSpan<T1, T2>,
+                                       HotOptions<T1, T2>,
+                                       HotContext<T1, T2>,
+                                       HotAdapter<T1, T2> > {
+    typedef GenericTracer<HotTracer<T1, T2>,
+            HotSpan<T1, T2>,
+            HotOptions<T1, T2>,
+            HotContext<T1, T2>,
+            HotAdapter<T1, T2> > Parent;
+
+    T1 & t1;
+    T2 & t2;
+  public:
+
+    HotTracer(T1& t1_, T2&t2_)
+        : t1(t1_)
+        , t2(t2_)
+    {
+    }
+
+    static void installImp(HotTracer*tracer)
+    {
+        s_tracer = tracer;
+    }
+    static void uninstallImp()
+    {
+        s_tracer = 0;
+    }
+    static HotTracer* instanceImp()
+    {
+        return s_tracer;
+    }
+
+    HotOptions<T1,T2>* makeSpanOptionsImp()
+    {
+        return new HotOptions<T1,T2>();
```

Comment:
Bad implementation. It should delegate etc. and then `HotOptions<T1,T2>` would proxy.